pywsitest/utils.py

```diff
@@ -1,23 +1,51 @@
 import re
+from typing import List, Tuple
 
 
 PATH_REGEX = re.compile(r"^\$\{(.*)\}$")
 
 
-def get_resolved_value(response: dict, path: str) -> object:
+def get_resolved_values(response: [list, dict], path: str) -> List[object]:
     """
-    Retrieves a value from a dictionary at a given path
+    Retrieves a list of values from a dictionary at a given path
 
     Parameters:
-        response (dict): The response to check against for a value
-        path (str): The path in the response to check for a value
+        response (list, dict): The response to check against for values
+        path (str): The path in the response to check for values
 
     Returns:
-        (object): The object at a given path, or None if the path can't be found
+        (list[object]): The list of objects at a given path, empty if the path can't be found
     """
-    resolved = response
-    for part in path.split("/"):
-        if part not in resolved:
-            return None
-        resolved = resolved[part]
+    resolved = [response]
+
+    for part in path.lstrip("/").split("/"):
+        # iterate to count rather than over objects as objects can be updated in-place
+        count = len(resolved)
+        for i in range(count):
+            current = resolved[i]
+            if isinstance(current, dict):
+                # update in-place with value at path or None
+                resolved[i] = current.get(part)
+            elif isinstance(current, list):
+                # index has been supplied, update in-place with object at index or None
+                if part:
+                    index, success = _to_int(part)
+                    resolved[i] = current[index] if success and len(current) > index else None
+                else:
+                    # append new resolved objects to end of the list
+                    for child in current:
+                        resolved.append(child)
+                    # set current object to None so it gets filtered out at the end of the loop
+                    resolved[i] = None
```

Comment:
another option might be to create a method that returns an enumerator and works on a clone, instead of the actual objects